### DISCUSSION TITLE
How do i set environment variables?

### DISCUSSION THREAD

**Discussion by m477r1x** [renovatebot/renovate#19350] [Request Help]
This is probably a silly question, but I've been looking through the confguration documentation and im seeing a few things which I'd like to implement, which can be configured either at runtim as a CLI argument, or as an environment variable, or in the `config.js` file. But i cannot seem to find the part of the documentation which demonstrates how to set environment variables?

I am using renovate as the hosted GitHub app, is it possible to set environment variables? Ideally what I want to do is configure a PAT so that Renvovate can look up dependencies in other private repos, but without having to put the PAT as plain text in the `renovate.json` file in every repo as this seems really unsecure.

**Comment by viceice** [comment]
You can't set environment variables for the hosted app. You can encrypt[^1] your secrets.

[^1]: https://docs.renovatebot.com/getting-started/private-packages/#encrypting-secrets
